Add holder name for new Faster Payments accounts

stejbac opened the pull request:
<!-- 
- make yourself familiar with the CONTRIBUTING.md if you have not already (https://github.com/bisq-network/bisq/blob/master/CONTRIBUTING.md)
- make sure you follow our [coding style guidelines][https://github.com/bisq-network/style/issues)
- pick a descriptive title
- provide some meaningful PR description below
- create the PR
- in case you receive a "Change request" and/or a NACK, please react within 30 days. If not, we will close your PR and it can not be up for compensation.
- After addressing the change request, __please re-request a review!__ Otherwise we might miss your PR as we tend to only look at pull requests tagged with a "review required".
-->

Use `excludeFromJsonDataMap` to add an optional owner full name property to Faster Payments (GBP) accounts, without affecting the trade JSON contract (for backwards compatibility). This is to fix #3976, that some banks have started checking the recipient name.

This uses the new map key: `public static final String HOLDER_NAME = "holderName";`

Make the full name a non-optional field for _new_ accounts, but display the details for old Faster Payments accounts in exactly the same way as before, only showing the extra field in the info popup, account and trade step views when it is present.

Also update the now misleading payment instructions in the buyer popup. With the change, it shows as:
![Screenshot from 2020-03-25 23-01-39](https://user-images.githubusercontent.com/54855381/77550995-b8346480-6eec-11ea-875c-33051313b3be.png)

Also add a popup to advise the user to recreate their Faster Payments account.

Make an info popup open in the take/create offer view, upon choosing to take or make a new offer, to instruct the user to recreate their old Faster Payments account with an owner full name (and preserved salt). Also show the popup upon manual selection of any old (i.e. missing full name) Faster Payments account from the trading account combo box, analogously to the ClearXchange (Zelle) warning popup logic.
![Screenshot from 2020-03-25 23-36-33](https://user-images.githubusercontent.com/54855381/77554904-9ee1e700-6ef1-11ea-99fd-972fb327ce0e.png)

I've also copied/overwritten the two new/updated English resource bundle strings into the non-English resource files, so that the translations stay in sync with the updated information. (As it is a UK-only payment method, the non-English translations are probably not all that important anyway.)

--

Finally, I removed some redundant null-handling of the `excludeFromJsonDataMap` field
and constructor parameters. This touched all the `PaymentAccountPayload` subclasses, which was quite a lot of files (although fairly trivial changes).

stejbac committed:
Remove redundant @Nullable from account payload excludeFromJsonDataMap

Since this map is final and every PaymentAccountPayload constructor
initialises it to something nonnull, the @Nullable field annotation is
redundant, so remove it.

Further simplify the code by passing an empty map into the constructor
directly from each subclass constructor, rather than mapping empties to
nulls from each 'proto.getExcludeFromJsonDataMap()' result, then mapping
nulls back to empties in the base constructor. This is safe, since
getExcludeFromJsonDataMap() always returns a nonnull map.

Finally, tidy the code slightly by replacing 'Charset.forName("UTF-8")'
with StandardCharsets.UTF_8 in each PaymentAccountPayload subclass.

stejbac committed:
Add holder name for new Faster Payments accounts

Use excludeFromJsonDataMap to add an optional owner full name property
to Faster Payments (GBP) accounts, without affecting the trade JSON
contract (for backwards compatibility). This is to fix #3976, that some
banks have started checking the recipient name.

Make the name a non-optional field for _new_ accounts, but display the
details for old Faster Payments accounts in exactly the same way as
before, only showing the extra field in the info popup, account and
trade step views when it is present.

Also update the now misleading payment instructions in the buyer popup.
TODO: Update translations (which probably aren't needed anyway).

stejbac committed:
Add popup to advise user to recreate Faster Payments account

Open an info popup in the take/create offer view, upon choosing to take
or make a new offer, to instruct the user to recreate their old Faster
Payments account with an owner full name (and preserved salt). Also show
the popup upon manual selection of any old (i.e. missing full name)
Faster Payments account from the trading account combo box, analogously
to the ClearXchange (Zelle) warning popup logic.

(Also eliminate slight differences between the private
'maybeShow[ClearXchange|FasterPayments]Warning' methods in TakeOfferView
and MutableOfferView, to make the code easier to deduplicate in future.)

ripcurlx commented:
@m52go Could you please review the wording?
I'll test it on Regtest after reviewing the code.

ripcurlx commented on desktop/src/main/java/bisq/desktop/components/paymentmethods/FasterPaymentsForm.java:
```diff
@@ -44,6 +44,10 @@
 
     public static int addFormForBuyer(GridPane gridPane, int gridRow,
                                       PaymentAccountPayload paymentAccountPayload) {
+        if (!((FasterPaymentsAccountPayload) paymentAccountPayload).getHolderName().isEmpty()) {
+            addCompactTopLabelTextField(gridPane, ++gridRow, Res.get("payment.account.owner"),
+                    ((FasterPaymentsAccountPayload) paymentAccountPayload).getHolderName());
+        }
```
If this field is shown during the trade process it creates unnecessary scrolling.
<img width="1312" alt="Bildschirmfoto 2020-03-26 um 16 16 33" src="https://user-images.githubusercontent.com/170962/77663881-d8603400-6f7d-11ea-918c-ed11a5557106.png">
I'd just change line 54-55 to
```
addCompactTopLabelTextField(gridPane, gridRow, 1, Res.get("payment.accountNr"),
                ((FasterPaymentsAccountPayload) paymentAccountPayload).getAccountNr());
```
which puts sort code and account nr next to each other.
<img width="1312" alt="Bildschirmfoto 2020-03-26 um 16 19 57" src="https://user-images.githubusercontent.com/170962/77663992-ff1e6a80-6f7d-11ea-8847-11f72f189964.png">
That would work for both cases.

stejbac replied:
OK, I'll apply the suggested change.

ripcurlx reviewed (changes requested):
NACK - please see my comment regarding the buyers UI during the trade process.
Also please only commit changes to the English source file all other files will be overwritten automatically by loading translations from Transifex before the next release.

ACK from a functionality perspective. Tested old and new Faster Payment Accounts in trades.

stejbac committed:
Improve popup text + layout of Faster Payments buyer form fields

Display the account number on the same row as the sort code in the trade
step view, to prevent scrolling with the extra name field (as suggested
in the code review).

(This also affects the layout of old accounts without the extra field.)

Also apply the suggested popup text simplifications.

m52go commented on core/src/main/resources/i18n/displayStrings.properties:
```diff
@@ -622,8 +622,9 @@ portfolio.pending.step2_buyer.halCashInfo.msg=You need to send a text message wi
-  by any of the banks.
+  The UK sort code and account number used to be sufficient for a Faster Payment transfer, but since then some banks \
+  have started verifying the receiver's name. For this reason, accounts created in old Bisq clients do not provide the \
+  trade partner with an owner full name. Please use trade chat to avoid any issues in such cases.
```
```suggestion
  Faster Payment accounts created in old Bisq clients do not provide the receiver's name, \
  so please use trade chat to obtain it (if needed).
```

stejbac replied:
OK, I'll apply that change. I think the text 

> Some banks might require the receiver's name ...

should possibly be changed to 

> Some banks might verify the receiver's name ...

with the suggested change, to emphasise that the issue is that they've started checking it. 

(Although my bank still accepts dummy names - I'm not sure how much longer, though.)

m52go commented on core/src/main/resources/i18n/displayStrings.properties:
```diff
@@ -3019,7 +3020,13 @@ If you have not fulfilled the above requirements you will lose your security dep
+  full name. Please consider recreating your account in Bisq to provide future {0} buyers with a recipient name.\n\n\
+  When recreating a Faster Payments account, take care to copy the account age verification salt from your old account \
+  and use the same sort code and account number. (The owner full name or account name need not be the same, however.) \
+  This preserves the public hash, so that the new account need not be aged or signed from scratch.
```
```suggestion
payment.fasterPayments.newRequirements.info=Some banks have started verifying the receiver''s \
  full name for Faster Payments transfers. Your current Faster Payments account does not specify a full name.\n\n \
  Please consider recreating your Faster Payments account in Bisq to provide future {0} buyers with a full name.\n\n\
  When you recreate the account, make sure to copy the precise sort code, account number, and account age \
  verification salt values from your old account to your new account. This will ensure your existing account's age and \
  signing status are preserved.
```

stejbac replied:
OK, I'll apply the suggested simplification.

m52go reviewed (commented):
Existing text was already good. I just made it more concise, since there's already so much text in the pop-ups.

I did not suggest edits across languages, in case anyone would like to propose edits to these edits. If these edits are acceptable, I can do so.

stejbac commented:
I've just done a force push to drop the last commit which made the unwanted translation changes, and replaced it with a commit to apply @ripcurlx's suggestion.

I just noticed the text change suggestions - will push those shortly.

stejbac commented:
I've just force pushed again to include the popup text simplifications in the last commit.
